packages/react-reconciler/src/ReactFiberUnwindWork.js

```diff
@@ -232,6 +234,12 @@ function throwException(
           );
           thenable.then(onResolveOrReject, onResolveOrReject);
 
+          // Use `updateQueue` field to signal that the Timeout needs to switch
+          // to the placeholder. We don't need an entire queue. Any non-null
+          // value works.
+          // $FlowFixMe - Intentionally using a value other than an UpdateQueue.
+          workInProgress.updateQueue = emptyObject;
```

Comment:
So this was bad enough when it was lose mode only but we shouldn't promote this hack to strict mode.